### DISCUSSION TITLE
Error Setting up Devices

### DISCUSSION THREAD

**Discussion by bellisnb** [fwestenberg/reolink_dev#269] [General]
Hi, 

Have installed the Reolink add on as per the the setup guide however when coming to add individual devices from my NVR getting the same error in the logs every time.  ERROR SETTING UP ENTRY BACK GARDEN FOR REOLINK_DEV.

This points me to yaml file "_init_.py" Line 77. looking at the async setup entry. Cannot see what the logs are saying is the issue? Any help much appreciated. Pasted log details below. 

Logger: homeassistant.config_entries
Source: helpers/network.py:116
First occurred: 9:52:19 PM (1 occurrences)
Last logged: 9:52:19 PM

Error setting up entry Back Garden for reolink_dev
Traceback (most recent call last):
  File "/config/custom_components/reolink_dev/__init__.py", line 77, in async_setup_entry
    push = hass.data[DOMAIN][base.push_manager]
KeyError: 'push_manager-ec71db1ea38f'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 304, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/config/custom_components/reolink_dev/__init__.py", line 86, in async_setup_entry
    await push.subscribe(base.event_id)
  File "/config/custom_components/reolink_dev/base.py", line 302, in subscribe
    get_url(self._hass, prefer_external=False),
  File "/usr/src/homeassistant/homeassistant/helpers/network.py", line 116, in get_url
    raise NoURLAvailableError
homeassistant.helpers.network.NoURLAvailableError

Thank You 

Nathan

**Comment by LuisRRodas** [comment]
Just had the same error. I am curious as well on how to fix this.

**Reply by fwestenberg** [reply]
Did you try this: https://github.com/fwestenberg/reolink_dev/discussions/222 ?

**Reply by LuisRRodas** [reply]
Thank yo so much for the reply. I did try the suggested and edited the configuration.yaml. I was not able to get it to work. It stated could not connect to the cameras.

**Reply by fwestenberg** [reply]
The NoURLAvailableError makes the subscription manager fail. So the url error comes first. Did you reboot after changing the URL? And what did you set the URL's to? In my case, only setting the internal URL is sufficient to make the subscription manager work.

**Reply by LuisRRodas** [reply]
I did restart after setting the internal url.
I have this added to the end of my configuration.yaml.
homeassistant:
  internal_url: http://192.168.1.26:8123
  external_url: https://myaliass.duckdns.org
